drivers/clock_control/clock_control_esp32.c

```diff
@@ -70,7 +72,7 @@ static int clock_control_esp32_on(const struct device *dev, clock_control_subsys
 {
 	enum clock_control_status status = clock_control_esp32_get_status(dev, sys);
 
-	if (status == CLOCK_CONTROL_STATUS_ON) {
+	if (status == CLOCK_CONTROL_STATUS_ON && !reset_reason_is_cpu_reset()) {
 		return -EALREADY;
```

Comment:
Error `EALREADY` is defined as "Operation already in progress". Are you sure this is the correct return value for the already started clock? Returning the `0` seems to be a more valid response to me.

Reply:
The decision to return an error instead of `0` comes from `test_double_start` testcase from `clock_control_api` test. The API expect an error in case we try to enable an already enabled subsystem.
We can discuss the appropriate return value, but a negative one is required for the driver to pass the tests.